File: Sources/OutfitPickerCore/Domain/Entities/FileEntry.swift
import Foundation

/// Internal model representing an individual outfit file with its context information.
public struct FileEntry: Equatable, Sendable {
    let filePath: String
    public let fileName: String
    private let _url: URL
    private let _categoryURL: URL

    public init(filePath: String) {
        self.filePath = filePath
        self._url = URL(filePath: filePath, directoryHint: .notDirectory)
        self.fileName = _url.lastPathComponent
        self._categoryURL = _url.deletingLastPathComponent()
    }

    public var categoryPath: String {
        _categoryURL.path(percentEncoded: false)
    }

    public var categoryName: String {
        _categoryURL.lastPathComponent
    }
}

File: outfitpicker-go/internal/domain/entities/file_entry.go
package entities

import "path/filepath"

// FileEntry represents an outfit file with its filesystem context.
type FileEntry struct {
	filePath     string
	categoryPath string
	FileName     string
}

// NewFileEntry creates a new file entry from a file path.
func NewFileEntry(filePath string) FileEntry {
	categoryPath := filepath.Dir(filePath)
	return FileEntry{
		filePath:     filePath,
		categoryPath: categoryPath,
		FileName:     filepath.Base(filePath),
	}
}

// CategoryPath returns the directory path containing this file.
func (f FileEntry) CategoryPath() string {
	return f.categoryPath
}

// CategoryName returns the name of the category directory.
func (f FileEntry) CategoryName() string {
	return filepath.Base(f.categoryPath)
}
